model/array_typespec.yaml

```diff
@@ -39,12 +39,12 @@
   - class_ref: index_typespec
     name: index typespec
     vpi: vpiIndexTypespec
-    type: typespec
+    type: ref_obj
```

Comment:
That's not good.
Within a typespec there should be a strong type, at least it should be a ref_typespec.
Actually everywhere it should be a ref_typespec, you should not have used ref_obj.